### DISCUSSION TITLE
Editor (WYSIWYG) component

### DISCUSSION THREAD

**Discussion by DelScipio** [zauberzeug/nicegui#1124] [Ideas / Feature Requests]
Quasar has a https://quasar.dev/vue-components/editor

It would be interesting to make it available in NiceGUI as can be a killer feature to develop multiple features.

**Comment by falkoschindler** [comment]
I'd love to see such an element in NiceGUI. The implementation should be similar to other value elements like `ui.input`.
Does anyone want to give it a try? A pull request would be very welcome.

**Reply by falkoschindler** [reply]
I just created a PR myself: https://github.com/zauberzeug/nicegui/pull/1651

**Comment by WolfgangFahl** [comment]
Just asked ChatGPT how to integrate monaco editor to Vue and the answer was:

_Create a Vue component to wrap the Monaco Editor. You'll need to import the monaco-editor in your script:_
```javascript
<template>
  <div ref="editor" class="editor-container"></div>
</template>

<script>
import * as monaco from 'monaco-editor'

export default {
  name: 'MonacoEditor',
  data() {
    return {
      editor: null
    }
  },
  mounted() {
    this.editor = monaco.editor.create(this.$refs.editor, {
      value: '// type your code...',
      language: 'javascript',
      theme: 'vs-dark',
    })
  },
  beforeDestroy() {
    if (this.editor) {
      this.editor.dispose()
    }
  },
}
</script>

<style scoped>
.editor-container {
  width: 100%;
  height: 100vh; /* Adjust the height as per your requirement */
}
</style>
```

_Note: Loading Monaco Editor can be heavy because of its size, so you might want to load it on demand (lazy load) or use a webpack plugin like monaco-editor-webpack-plugin to optimize it._

I have seen https://github.com/zauberzeug/nicegui/tree/main/examples/custom_vue_component but don't know how the conversion from the above javascript and the loading procedure would be done ...

see also https://stackoverflow.com/questions/68846422/how-to-integrate-monaco-editor-with-electron-vue
